### PR TITLE
docs: Add section about project's rationale

### PR DESCRIPTION
**Pull request by jakubjanecek**

### CODE REVIEW

File: docs/pureconfig.md [hunk 1]
```diff
@@ -20,9 +20,9 @@ import zio.Task
 final case class ServerConfiguration(listenAddress: String, listenPort: Int)
 
 implicit val serverConfigurationReader: ConfigReader[ServerConfiguration] = deriveReader
-// serverConfigurationReader: ConfigReader[ServerConfiguration] = pureconfig.generic.DerivedConfigReader1$$anon$3@7d4e424e
+// serverConfigurationReader: ConfigReader[ServerConfiguration] = pureconfig.generic.DerivedConfigReader1$$anon$3@3c04ddda
```
Comment (tomasherman) [inline docs/pureconfig.md]: really? :)

File: example/src/main/mdoc/rationale.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Rationale
```
Comment (tomasherman) [inline example/src/main/mdoc/rationale.md]: why is this here twice?

File: docs/pureconfig.md (hunk 1 above)
Comment (jakubjanecek) [reply docs/pureconfig.md]: That's generated document with output from console. I'll probably redo this somehow but it is not important for now and it serves as some documentation...

File: example/src/main/mdoc/rationale.md (hunk 2 above)
Comment (jakubjanecek) [reply example/src/main/mdoc/rationale.md]: This document is the source, the other one is generated out of this by mdoc. This document does not contain any code so it could be here only once but I like to keep it consistent and have all the sources at single place.
